modules/katello_repository_set.py

```diff
@@ -0,0 +1,188 @@
+            - Organization name that the lifecycle environment is in
+        required: true
+    state:
+        description:
+            - Whether the repositories are enabled or not
+        required: true
+        choices:
+            - 'enabled'
+            - 'disabled'
+'''
+
+EXAMPLES = '''
+- name: "Enable RHEL 7 RPMs repositories"
+  katello_lifecycle_environment:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    organization: "Default Organization"
+    product: "Red Hat Enterprise Linux Server"
+    repositories:
+    - releasever: "7.0"
+      basearch: "x86_64"
+    - releasever: "7.1"
+      basearch: "x86_64"
+    - releasever: "7.2"
+      basearch: "x86_64"
+    - releasever: "7.3"
+      basearch: "x86_64"
+    state: enabled
+'''
+
+RETURN = '''# '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        find_organization,
+        find_product,
+        find_repository_set,
+        ping_server,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+
+def get_desired_repos(desired_substitutions, available_repos):
+    desired_repos = []
+    for sub in desired_substitutions:
+        desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
+    return desired_repos
+
+
+def repository_set(module, name, organization, product, state, repositories=[]):
+    changed = False
+    organization = find_organization(module, organization)
+    product = find_product(module, product, organization)
+    repo_set = find_repository_set(module, name=name, product=product)
+
+    available_repos = repo_set.available_repositories()['results']
+    current_repos = map(lambda r: r.read(), repo_set.repositories)
+    desired_repos = get_desired_repos(repositories, available_repos)
+
+    available_repo_names = set(map(lambda r: r['repo_name'], available_repos))
+    current_repo_names = set(map(lambda r: r.name, current_repos))
+    desired_repo_names = set(map(lambda r: r['repo_name'], desired_repos))
+
+    if len(desired_repo_names - available_repo_names) > 0:
+        module.fail_json(msg="Desired repositories are not available on the repository set {}. Desired: {} Available: {}"
+                         .format(name, desired_repo_names, available_repo_names))
+
+    if state == 'enabled':
+        for repo in desired_repo_names - current_repo_names:
+            repo_to_enable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.enable(data=repo_to_enable['substitutions'])
+            changed = True
+    else:
+        for repo in current_repo_names & desired_repo_names:
+            repo_to_disable = (r for r in available_repos if r['repo_name'] == repo).next()
+            repo_set.disable(data=repo_to_disable['substitutions'])
+            changed = True
```

Comment:
Yes, you get the server response wrapped in a ansible error:

```sh
$ ansible-playbook test/demo/repository-set.yml
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [localhost] ***************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Disable RHEL 7 RPMs repositories] ****************************************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "WARNING:nailgun.client:Received HTTP 500 response: {\"displayMessage\":\"Repository cannot be deleted since it has already been included in a published Content View. Please delete all Content View versions containing this repository before attempting to delete it.\",\"errors\":[\"Repository cannot be deleted since it has already been included in a published Content View. Please delete all Content View versions containing this repository before attempting to delete it.\"]}\nTraceback (most recent call last):\n  File \"/tmp/ansible_3j3KkO/ansible_module_katello_repository_set.py\", line 187, in <module>\n    main()\n  File \"/tmp/ansible_3j3KkO/ansible_module_katello_repository_set.py\", line 181, in main\n    module.fail_json(msg=e)\n  File \"/tmp/ansible_3j3KkO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1993, in fail_json\n  File \"/tmp/ansible_3j3KkO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1973, in _return_formatted\n  File \"/tmp/ansible_3j3KkO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 414, in remove_values\n  File \"/tmp/ansible_3j3KkO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 414, in <genexpr>\n  File \"/tmp/ansible_3j3KkO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 425, in remove_values\nTypeError: Value of unknown type: <class 'requests.exceptions.HTTPError'>, 500 Server Error: Internal Server Error for url: https://katello.example.com/katello/api/v2/products/6/repository_sets/2456/disable\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}
        to retry, use: --limit @/home/vagrant/foreman-ansible-modules/test/demo/repository-set.retry

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```